src/SmartComponents/RunTaskModal/InputParameters.js

```diff
@@ -1,75 +1,72 @@
-
-  const updateParams = (value) => {
-    setParamValue(value);
-    setDefinedParameters((prevState) => {
-      const newState = prevState.map((param) => {
-        if (param.key === parameter.key) {
-          return { key: param.key, value: value };
-        } else {
-          return param;
-        }
-      });
-      return newState;
-    });
+export const InputParameter = ({ parameter, updateParameters }) => {
+  const getInputElementType = (parameter) => {
+    const inputElementType = getParameterFormInputElementType(parameter);
+    if (inputElementType === 'textbox') {
+      return (
+        <ParameterTextbox
+          parameter={parameter}
+          updateParameters={updateParameters}
+        />
+      );
+    } else if (inputElementType === 'checkbox') {
+      return (
+        <ParameterCheckbox
+          parameter={parameter}
+          updateParameters={updateParameters}
+        />
+      );
+    } else {
+      return (
+        <ParameterDropdown
+          parameter={parameter}
+          updateParameters={updateParameters}
+        />
+      );
+    }
   };
 
   return (
-    <Form className="pf-v5-u-pb-lg">
-      <FormGroup
-        label={parameter.title || parameter.key}
-        isRequired={parameter.required}
-        type="text"
-        fieldId="name"
-      >
-        <FormSelect
-          value={paramValue}
-          onChange={(_event, value) => updateParams(value)}
-          aria-label={`Edit parameter ${parameter.key} value field`}
-        >
-          {parameter.values.map((value, index) => (
-            <FormSelectOption key={index} value={value} label={value} />
-          ))}
-        </FormSelect>
-        <FormHelperText>
-          <HelperText>
-            <HelperTextItem>{parameter.description}</HelperTextItem>
-          </HelperText>
-        </FormHelperText>
-      </FormGroup>
-    </Form>
+    <Form className="pf-v5-u-pb-lg">{getInputElementType(parameter)}</Form>
   );
 };
 
 InputParameter.propTypes = {
   parameter: propTypes.object,
-  setDefinedParameters: propTypes.func,
+  updateParameters: propTypes.func,
 };
 
 const InputParameters = ({ parameters, setDefinedParameters }) => {
-  return parameters.map((param) => {
-    return (
-      <InputParameter
-        key={param.key}
-        aria-label={param.key}
-        parameter={param}
-        setDefinedParameters={setDefinedParameters}
-      />
-    );
-  });
+  const updateParameters = (parameter, newValue) => {
+    setDefinedParameters((prevState) => {
+      return prevState.map((param) => {
+        if (param.key === parameter.key) {
+          return { key: param.key, value: newValue };
+        } else {
+          return param;
+        }
+      });
+    });
+  };
+  return parameters
+    .sort((a, b) => a.index - b.index) // sort the parameters on their index values
+    .map((param) => {
+      return (
+        <InputParameter
+          key={param.key}
+          aria-label={param.key}
```

Comment:
The key prop is needed to satisfy the linter regarding an iterator having a key.  The aria-label probably isn't needed because each sub-component has its own aria-label.